scripts/setup_build/setup_build.vcxproj

```diff
@@ -132,6 +136,9 @@
       <SubSystem>Console</SubSystem>
       <GenerateDebugInformation>true</GenerateDebugInformation>
     </Link>
+    <PreBuildEvent>
+      <Command>powershell -NonInteractive -ExecutionPolicy Unrestricted $(SolutionDir)scripts\generate-commitid.ps1 $(OutDir)</Command>
+    </PreBuildEvent>
```

Comment:
Does this really need to be rerun every time one builds even if nothing has changed?  Won't this generate a new (meaning updated timetstamp on file) git_common_id.h file that will then cause other things to be rebuilt?

Seems like we ought to update the script to only regenerate the file if the commit id has changed.

Reply:
Good idea.